internal/provider/edgegw/nat_rule_schema.go

```diff
@@ -0,0 +1,261 @@
+				Resource: &schemaR.StringAttribute{
+					Required: true,
+					// TODO - Validator of IP or IP/CIDR
+				},
+				DataSource: &schemaD.StringAttribute{
+					Computed: true,
+				},
+			},
+			"internal_address": superschema.SuperStringAttribute{
+				Common: &schemaR.StringAttribute{
+					MarkdownDescription: "The internal address for the NAT Rule. This must be supplied as a single IP or Network CIDR. For a DNAT rule, this is the internal IP address for incoming traffic. For an SNAT rule, this is the internal IP Address for outgoing traffic. For a REFLEXIVE rule, these are the internal IPs. These IPs are typically the Private IPs that are allocated to workloads.",
+				},
+				Resource: &schemaR.StringAttribute{
+					Required: true,
+					// TODO - Validator of IP or IP/CIDR
+				},
+				DataSource: &schemaD.StringAttribute{
+					Computed: true,
+				},
+			},
+			// Not implemented because App_Port_profile for nat rule allow only one port list and one protocol - no real utility
+			// "app_port_profile_id": superschema.SuperStringAttribute{
+			// 	Common: &schemaR.StringAttribute{
+			// 		MarkdownDescription: "Application Port Profile ID to which the rule applies.",
+			// 	},
+			// 	Resource: &schemaR.StringAttribute{
+			// 		Optional: true,
+			// 	},
+			// 	DataSource: &schemaD.StringAttribute{
+			// 		Computed: true,
+			// 	},
+			// },
+			"dnat_external_port": superschema.SuperStringAttribute{
+				Common: &schemaR.StringAttribute{
+					MarkdownDescription: "This represents the external port number or port range when doing DNAT port forwarding from external to internal. If not specify, all ports are translated",
+				},
+				Resource: &schemaR.StringAttribute{
+					Optional: true,
+					Validators: []validator.String{
+						fstringvalidator.RequireIfAttributeIsOneOf(path.MatchRoot("rule_type"), []attr.Value{types.StringValue("DNAT"), types.StringValue("NO_DNAT")}),
+						fstringvalidator.NullIfAttributeIsOneOf(path.MatchRoot("rule_type"), []attr.Value{types.StringValue("SNAT"), types.StringValue("NO_SNAT")}),
+					},
+				},
+				DataSource: &schemaD.StringAttribute{
+					Computed: true,
+				},
+			},
+			"snat_destination_address": superschema.SuperStringAttribute{
+				Common: &schemaR.StringAttribute{
+					MarkdownDescription: "The destination addresses to match in the SNAT Rule. This must be supplied as a single IP or Network CIDR. Providing no value for this field results in match with ANY destination network.",
+				},
+				Resource: &schemaR.StringAttribute{
+					Optional: true,
+					Validators: []validator.String{
+						fstringvalidator.RequireIfAttributeIsOneOf(path.MatchRoot("rule_type"), []attr.Value{types.StringValue("SNAT"), types.StringValue("NO_SNAT")}),
+						fstringvalidator.NullIfAttributeIsOneOf(path.MatchRoot("rule_type"), []attr.Value{types.StringValue("DNAT"), types.StringValue("NO_DNAT")}),
+					},
+				},
+				DataSource: &schemaD.StringAttribute{
+					Computed: true,
+				},
+			},
```

Comment:
Same `dnat_external_port`

Reply:
Fix, same also, i retired the require validator